@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -363,13 +387,50 @@ export class VisibleTileSet {
             }
         });
 
+        let minElevation = EarthConstants.MAX_ELEVATION;
+        let maxElevation = EarthConstants.MIN_ELEVATION;
         this.dataSourceTileList.forEach(renderListEntry => {
             const dataSource = renderListEntry.dataSource;
             const cache = this.m_dataSourceCache.get(dataSource.name);
             if (cache !== undefined) {
                 cache.tileCache.shrinkToCapacity();
             }
+            // Calculate min/max elevation from every data source tiles,
+            // datasources without elevationRangeSource will contribute to
+            // values with zero levels for both elevations.
+            const tiles = renderListEntry.visibleTiles;
+            tiles.forEach(tile => {
+                minElevation = Math.min(minElevation, tile.minElevation);
+                maxElevation = Math.max(maxElevation, tile.maxElevation);
+            });
         });
+        // Update elevation ranges stored.
+        if (
+            minElevation !== this.m_elevationRange.minElevation ||
+            maxElevation !== this.m_elevationRange.maxElevation
+        ) {
+            this.m_elevationRange.minElevation = minElevation;
+            this.m_elevationRange.maxElevation = maxElevation;
+        }
+        this.m_elevationRange.calculationStatus = visibleTileKeysResult.allBoundingBoxesFinal
+            ? CalculationStatus.FinalPrecise
+            : CalculationStatus.PendingApproximate;
+
+        // If clip planes evaluator depends on the tiles elevation re-calculate
+        // frustum planes and update the camera near/far plane distances.
+        const clipPlanesEvaluator = this.options.clipPlanesEvaluator;
+        let viewRangesChanged: boolean = false;
+        if (clipPlanesEvaluator instanceof ElevationBasedClipPlanesEvaluator) {
```

Comment:
This is quite ugly. Please add a setter for min and max elevation to the base ClipPlanesEvaluator interface. The ClipPlanesEvaluator will then decide whether it makes use of these values or not.